app/_config/security.yml

```diff
@@ -0,0 +1,20 @@
+---
+Name: appsecurity
+After: '#cwpsecurity'
+Except:
+  environment: dev
+---
+# Enable HTTP Strict Transport Security (HSTS), with a short max-age default.
+# Once you're confident with the setting in production,
+# it is encouraged to increase the max-age.
+# See https://www.cwp.govt.nz/developer-docs/en/2/working_with_projects/security/
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+CWP\Core\Control\InitialisationMiddleware:
+  strict_transport_security: 'max-age: 300'
+# Configure SSL redirection for every route (override default ForceSSLPatterns).
+# Required for the HSTS configuration above to take effect.
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\Control\Middleware\CanonicalURLMiddleware:
+    properties:
+      ForceSSL: true
+      ForceSSLPatterns: []
```

Comment:
This won't clear previously set values, since arrays are merge in YAML - was that the intention?

Reply:
Good catch! I've messed up my verification - ran locally in `test` mode, which already attempts to do the same thing to ensure basic auth is always sent over SSL. Unlike my code (`ForceSSLPatterns: []`), it unsets the patterns property (`ForceSSLPatterns: null`), which I've missed because of this overlapping config in `test` mode. See https://github.com/silverstripe/cwp-core/blob/master/_config/security.yml#L76

Fixed now.